.pre-commit-config.yaml

```diff
@@ -0,0 +1,50 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v5.0.0
+  hooks:
+  - id: trailing-whitespace
+  - id: end-of-file-fixer
+    exclude: |
+      (?x)^(
+        config/README.md|
+        cluster/README.md
+      )$
+  - id: mixed-line-ending
+  - id: check-yaml
+    args:
+    - --allow-multiple-documents
+  - id: check-added-large-files
+  - id: check-executables-have-shebangs
+  - id: check-shebang-scripts-are-executable
+  - id: check-merge-conflict
+  - id: no-commit-to-branch
+- repo: https://github.com/antonbabenko/pre-commit-terraform
+  rev: v1.96.2
+  hooks:
+  - id: terraform_fmt
+  - id: terraform_validate
+  - id: terraform_tflint
+  # - id: terraform_trivy
+  #   args:
+  #   - --args=--severity CRITICAL
+  #   - --args=--skip-dirs="**/.terraform"
+  # - id: terraform_checkov
+  #   args:
+  #   - --args=--quiet
+  #   - --args=--skip-check CKV_AWS_23,CKV_AWS_355,CKV_AWS_290 # Temporarly skip these checks, Helm chart testing isn't needed for the repo
```

Comment:
I saw this was mentioned to be enabled in the next step. Can you consider including this in the same PR?

Reply:
Enabling this will return around 30 errors that should be fixed, some of them are easy to fix, the others I can't fix them and I'd need your help for them. I've also created #37 and #38 that should be fixed for the tests to be passed. 
This PR is already rather big, it will be much bigger if we want to fix everything in on PR.

Reply:
Excluding #37 and #38, here is the list of issues that checkov finds:
```
Check: CKV_TF_1: "Ensure Terraform module sources use a commit hash"
Check: CKV_TF_1: "Ensure Terraform module sources use a commit hash"
Check: CKV_TF_1: "Ensure Terraform module sources use a commit hash"
Check: CKV_TF_1: "Ensure Terraform module sources use a commit hash"
Check: CKV_TF_1: "Ensure Terraform module sources use a commit hash"
Check: CKV_TF_1: "Ensure Terraform module sources use a commit hash"
Check: CKV_TF_1: "Ensure Terraform module sources use a commit hash"
Check: CKV_AWS_382: "Ensure no security groups allow egress from 0.0.0.0:0 to port -1"
Check: CKV_TF_1: "Ensure Terraform module sources use a commit hash"
Check: CKV_TF_1: "Ensure Terraform module sources use a commit hash"
Check: CKV_TF_1: "Ensure Terraform module sources use a commit hash"
Check: CKV_TF_1: "Ensure Terraform module sources use a commit hash"
Check: CKV2_AWS_61: "Ensure that an S3 bucket has a lifecycle configuration"
Check: CKV2_AWS_30: "Ensure Postgres RDS as aws_db_instance has Query Logging enabled"
Check: CKV2_AWS_65: "Ensure access control lists for S3 buckets are disabled"
Check: CKV_AWS_144: "Ensure that S3 bucket has cross-region replication enabled"
Check: CKV2_AWS_39: "Ensure Domain Name System (DNS) query logging is enabled for Amazon Route 53 hosted zones"
Check: CKV2_AWS_62: "Ensure S3 buckets should have event notifications enabled"
Check: CKV2_AWS_6: "Ensure that S3 bucket has a Public Access block"
Check: CKV2_AWS_60: "Ensure RDS instance with copy tags to snapshots is enabled"
Check: CKV_AWS_21: "Ensure all data stored in the S3 bucket have versioning enabled"
Check: CKV2_AWS_38: "Ensure Domain Name System Security Extensions (DNSSEC) signing is enabled for Amazon Route 53 public hosted zones"
Check: CKV_AWS_18: "Ensure the S3 bucket has access logging enabled"
Check: CKV_AWS_145: "Ensure that S3 buckets are encrypted with KMS by default"
Check: CKV2_AWS_19: "Ensure that all EIP addresses allocated to a VPC are attached to EC2 instances"
Check: CKV2_GHA_1: "Ensure top-level permissions are not set to write-all"
```

Reply:
Alright. Will have a look and see if I can contribute, thinking 30 doesn't sound extremely large to me and maybe some of them can also be safely ignored (e.g., the rules about S3 are definitely not mandatory and should be applied case by case).

Reply:
Indeed, some of them can be skipped.